File: frontend/src/services/transactionService.js
import API from "./api";

export const getTransactions = () =>
  API.get("/transactions");

export const addTransaction = (transaction) =>
  API.post("/transactions", transaction);

export const importTransaction = (transaction) =>
  API.post("/transactions/import", transaction);

export const deleteTransaction = (id) =>
  API.delete(`/transactions/${id}`);

export const updateTransaction = (id, transaction) =>
  API.put(`/transactions/${id}`, transaction);

File: backend/src/main/java/com/pennypilot/service/TransactionService.java
package com.pennypilot.service;

import com.pennypilot.entity.Transaction;

import java.util.List;

public interface TransactionService {

    Transaction addTransaction(Transaction transaction);

    Transaction addImportedTransaction(Transaction transaction);

    List<Transaction> getAllTransactions();

    void deleteTransaction(Long id);

    Transaction updateTransaction(Long id, Transaction transaction);
}